docs/apm.asciidoc

```diff
@@ -0,0 +1,207 @@
++
+First we need to store the certificate authority of the Elasticsearch cluster in a secret:
++
+[source,yaml]
+----
+kubectl create secret generic es-ca --from-file=tls.crt=elasticsearch-ca.crt
+----
+In the above command the file `elasticsearch-ca.crt` must contain the CA certificate of the Elasticsearch cluster you want to use.
++
+. You can then mount this secret using the pod template and reference the file in the `config` of the APM server.
++
+Here is a complete example with a password stored in the Keystore as it is described in the previous section:
++
+[source,yaml]
+----
+apiVersion: apm.k8s.elastic.co/v1alpha1
+kind: ApmServer
+metadata:
+  name: apm-server-quickstart
+  namespace: default
+spec:
+  version: "7.2.0"
+  nodeCount: 1
+  secureSettings:
+    secretName: apm-secret-settings
+  config:
+    output:
+      elasticsearch:
+        hosts: ["my-own-elasticsearch-cluster:9200"]
+        username: elastic
+        password: "${ES_PASSWORD}"
+        protocol: "https"
+        ssl.certificate_authorities: ["/usr/share/apm-server/config/elasticsearch-ca/tls.crt"]
+  podTemplate:
+    spec:
+      containers:
+      - name: apm-server
+        volumeMounts:
+        - mountPath: /usr/share/apm-server/config/elasticsearch-ca
+          name: elasticsearch-ca
+          readOnly: true
+      volumes:
+      - name: elasticsearch-ca
+        secret:
+          defaultMode: 420
+          optional: false
+          secretName: es-ca # This is the secret that holds the Elasticsearch CA cert
+----
+
+==== TLS Certificates
+
+By default the operator manages a private CA and generates a self-signed certificate used to secure the communication between APM agents and the server.
+
+This behavior and the relevant configuration is identical to what is done for Elasticsearch and Kibana. You can refer to link:k8s-accessing-elastic-services.html#k8s-setting-up-your-own-certificate[this documentation] to understand how to use your own certificate to configure the TLS endpoint of the APM server.
+
+=== Connecting to the APM server
+
+==== APM Server service
+
+The APM server is exposed with a Service, you can refer to link:k8s-accessing-elastic-services.html[this documentation] to understand how to access it.
+
+You can retrieve the list of all the APM Services with following command:
+
+[source,sh]
+----
+kubectl get service --selector='common.k8s.elastic.co/type=apm-server'
+----
+
+[source,sh]
+----
+NAME                             TYPE        CLUSTER-IP   EXTERNAL-IP   PORT(S)    AGE
+apm-server-quickstart-apm-http   ClusterIP   10.0.1.252   <none>        8200/TCP   154m
+----
+
+==== APM Server secret token
+
+The operator generates an authorization token that the agents must send to authenticate themselves to the APM server.
+
+This token is stored in a secret and can be retrieved with the following command:
+
+[source,sh]
+----
+kubectl get secret/apm-server-quickstart-apm-token  -o json | jq -r '.data| .["secret-token"]' | base64 -D
```

Comment:
Use go-template instead to avoid reliance on external tools?
``
kubectl get secret/apm-server-quickstart-apm-token  -o go-template='{{index .data "secret-token" | base64decode}}'
```